### PR TITLE
Step 6 of walkthrough- limit and delete comments

### PR DESCRIPTION
**Pull request by indhu-gunda**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for deleting tasks. */
+@WebServlet("/delete-comment")
+public class DeleteCommentServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long id = Long.parseLong(request.getParameter("id"));
```
Comment (BearBurg) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java]: Consider wrapping it with `try` `catch` for better fault tolerance. You can add this error handling if you have extra time.
For example, you can catch the exception and return 400 with some error message.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    Key commentEntityKey = KeyFactory.createKey("Comment", id);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.delete(commentEntityKey);
+  }
+}
```
Comment (BearBurg) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java]: Please add a new line at the end of this file.

File: portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java [hunk 3]
```diff
@@ -34,25 +38,19 @@
 @WebServlet("/list-comments")
 public class ListCommentsServlet extends HttpServlet {
 
+  private static final Gson gson = new Gson();
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int maxNumComments = Integer.parseInt(request.getParameter("max"));
```
Comment (BearBurg) [inline portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java]: Same as before, you can set the `maxNumComments` to a default value or return an empty list if the parameter does not exist or is not valid.

**Review by BearBurg — APPROVED** [review]
LGTM

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java (hunk 1 above)
Comment (indhu-gunda) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java]: fixed

File: portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java (hunk 3 above)
Comment (indhu-gunda) [reply portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java]: fixed

**Pull request merged**